aiida/orm/nodes/data/array/cube.py

```diff
@@ -0,0 +1,246 @@
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        if comment:
+            self.comment = comment
+        if voxel:
+            self.voxel = voxel
+        if origin:
+            self.origin = origin
+        if atomic_numbers:
+            self.atomic_numbers = atomic_numbers
+        if atomic_charges:
+            self.atomic_charges = atomic_charges
+        if atomic_coordinates:
+            self.atomic_coordinates = atomic_coordinates
+        if data:
+            self.data = data
+        if data_units:
+            self.data_units = data_units
+
+    @property
+    def comment(self):
+        """First two lines of the cube file that contain the comment"""
+        return self.get_attribute('comment')
+
+    @comment.setter
+    def comment(self, comment):
+        if isinstance(comment, (list, tuple)) and len(comment) == 2:
+            if isinstance(comment[0], str) and isinstance(comment[1], str):
+                self.set_attribute('comment', comment)
+                return
+        raise ValueError(f'The attibue `comment` must be a list or a tuple containing two strings, got: {comment}')
+
+    @property
+    def voxel(self):
+        """Three vectors defining a voxel  (bohr)"""
+        return self.get_array('voxel')
+
+    @voxel.setter
+    def voxel(self, voxel):
+        self.set_array('voxel', voxel)
+
+    @property
+    def origin(self):
+        """Position of the origin of the volumeric data (bohr)."""
+        return self.get_array('origin')
+
+    @origin.setter
+    def origin(self, origin):
+        self.set_array('origin', origin)
+
+    @property
+    def atomic_numbers(self):
+        """Atomic numbers in the structure."""
+        return self.get_array('atomic_numbers')
+
+    @atomic_numbers.setter
+    def atomic_numbers(self, atomic_numbers):
+        self.set_array('atomic_numbers', atomic_numbers)
+
+    @property
+    def atomic_charges(self):
+        """Atomic charges in the structure."""
+        return self.get_array('atomic_charges')
+
+    @atomic_charges.setter
+    def atomic_charges(self, atomic_charges):
+        self.set_array('atomic_charges', atomic_charges)
+
+    @property
+    def atomic_coordinates(self):
+        """Atomic coordinates in the structure."""
+        return self.get_array('atomic_coordinates')
+
+    @atomic_coordinates.setter
+    def atomic_coordinates(self, atomic_coordinates):
+        self.set_array('atomic_coordinates', atomic_coordinates)
+
+    @property
+    def data(self):
+        """3D array containing the mesh data."""
+        return self.get_array('data')
+
+    @data.setter
+    def data(self, data):
+        self.set_array('data', data)
+
+    @property
+    def data_units(self):
+        """Units of the mesh data."""
+        return self.get_attribute('data_units')
+
+    @data_units.setter
+    def data_units(self, data_units):
+        self.set_attribute('data_units', data_units)
+
+    def read(self, file, data_units=None):
```

Comment:
What is the argument for making this an instance method as opposed to a class method? Why would you want to have to construct an instance first, if after all what you are doing here _is_ constructing a new instance